### ISSUE TITLE
Default Hot reload indicator position is problematic

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/uno#18053]
### What would you like to be added

Currently hot reload indicator is on the top left, which often collides with navigation - back button, hamburger menu, etc. - see https://github.com/unoplatform/uno/issues/18024.

A more suitable location would be top center (similar to WPF) or top right. It is also worth considering making the UI semi-opaque unless actively interacted with pointer.

WPF:

![image](https://github.com/user-attachments/assets/5f92cb9c-ba58-40d6-90b3-46e4d7636f70)

Flutter:

![image](https://github.com/user-attachments/assets/2dbbb8eb-7ab6-41a9-b79a-b39aca93b472)

### Why is this needed

UX

**Comment by nickrandolph** [comment]
I think preference here would be to align right, to avoid conflicts with any center toolbar that VS sometimes displays

**Comment by Youssef1313** [comment]
Right-aligned may be problematic for Right-To-Left apps.

**Comment by MartinZikmund** [comment]
I noticed the `DiagnosticOverlay` already has support for alternative placement, can we change the default to just be top right? @nickrandolph 

In all apps using `NavigationView` it initially overlaps the hamburger menu, so it just has to be moved away

![Image](https://github.com/user-attachments/assets/7442f6ed-4fd8-42ae-aa6c-5a29b83bb85a)

**Comment by nickrandolph** [comment]
Yes agreed. Part of a wider issue related to the UX of the toolbar. In short term we can move the default to right side
